extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/HttpServerCommonHandlers.java

```diff
@@ -133,7 +133,7 @@ public void handle(RoutingContext event) {
                                 .handler(new Handler<RoutingContext>() {
                                     @Override
                                     public void handle(RoutingContext event) {
-                                        event.response().headers().setAll(headers);
+                                        event.response().headers().addAll(headers);
```

Comment:
I'm not too familiar with this code, but just in case, can this cause some headers (not even CORS related ) being duplicated ?

Reply:
`add` does indeed add a header, not replace it, so it could theoretically duplicate a value. 

But I don't really think it's something this could should handle, I believe it's up to the user to ensure this doesn't happen.

Reply:
We need to check how addAll handle duplicates. Headers are a multimap, so we may have multiple value for a single header. I'm not sure it's what we want in this case (but we need to verify).

Also, this should be considered as a breaking change.

Reply:
> Also, this should be considered as a breaking change.

Yeah, I agree with this. It's breaking, although I argue that the intent was to do what is being done now.

Reply:
I believe the reason behind `set` was this duplicate issue. 
With this PR, we cannot remove a value anymore or decide when we want a single or multiple values of a header (even if I agree that having multiple values of the same header is rarely used, this PR may introduce that case a lot more).